exam/helpers.py

```diff
@@ -160,3 +160,22 @@ def __call__(self, *args, **kwargs):
                 return return_value
 
         raise TypeError('Unknown effect for: %r, %r' % (args, kwargs))
+
+
+class sequential_effect(effect):
+    '''Subclass from exam.helpers.effect, allowing for sequential calls'''
+    
+    def __call__(self, *args, **kwargs):
+        this_call = self.call_class(*args, **kwargs)
+        for call_obj, return_value in self:
```

Comment:
I *think* that since it raises TypeError in case of no match on the first iteration, if the head doesn't match the call it will fail... This will need some more test cases.